src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4231,7 +4231,7 @@ public bool ConnectAsync(SocketAsyncEventArgs e)
                     {
                         InternalBind(new IPEndPoint(IPAddress.Any, 0));
                     }
-                    else
+                    else if (endPointSnapshot.AddressFamily == AddressFamily.InterNetworkV6)
```

Comment:
This is breaking app-compat with Desktop. 
Please explain why this is needed and how was this tested. Also, given that you are changing the behavior for non-IP sockets, we'll need tests to ensure that those still work the same way as on Desktop.